web/modules/custom/joinup_federation/src/Plugin/pipeline/Step/UserSelectionFilter.php

```diff
@@ -172,14 +172,15 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
     $options = $default_value = [];
     foreach ($entities_per_category as $category_id => $entities) {
       foreach ($entities as $id => $label) {
-        $options[$id] = [
+        $data = [
           'solution' => $label,
           'info' => $this->getInfo($id, $category_id),
           '#attributes' => ['data-drupal-federation-category' => $category_id],
         ];
-        $default_value[$id] = $category_id !== 'blacklisted' ? $id : NULL;
+        $options[$id] = $data;
+        $default_value[$id] = in_array($category_id, ['blacklisted', 'invalid_collection']) ? NULL : $id;
+        $options[$id]['#disabled'] = $category_id === 'invalid_collection';
```

Comment:
The `#disabled` property can be added directly in the initial array, like `#attributes`.